app/uk/gov/hmrc/exports/services/reversemapping/declaration/items/SingleItemParser.scala

```diff
@@ -17,19 +17,45 @@
 package uk.gov.hmrc.exports.services.reversemapping.declaration.items
 
 import java.util.UUID
+
 import scala.xml.NodeSeq
+
 import javax.inject.Inject
-import uk.gov.hmrc.exports.models.declaration.ExportItem
+import uk.gov.hmrc.exports.models.declaration.YesNoAnswer.YesNoAnswers
+import uk.gov.hmrc.exports.models.declaration._
+import uk.gov.hmrc.exports.services.reversemapping.MappingContext
 import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser
-import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser.XmlParserResult
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser._
 import uk.gov.hmrc.exports.services.reversemapping.declaration.XmlTags._
-import uk.gov.hmrc.exports.services.reversemapping.MappingContext
 
 class SingleItemParser @Inject()(procedureCodesParser: ProcedureCodesParser) extends DeclarationXmlParser[ExportItem] {
 
   override def parse(itemXml: NodeSeq)(implicit context: MappingContext): XmlParserResult[ExportItem] =
-    procedureCodesParser.parse(itemXml).map {
-      val sequenceId = (itemXml \ SequenceNumeric).text.toInt
-      ExportItem(UUID.randomUUID().toString, sequenceId, _)
+    procedureCodesParser.parse(itemXml).map { procedureCodes =>
+      val domesticDutyTaxParties = itemXml \ DomesticDutyTaxParty
+
+      ExportItem(
+        id = UUID.randomUUID().toString,
+        sequenceId = (itemXml \ SequenceNumeric).text.toInt,
+        procedureCodes = procedureCodes,
+        fiscalInformation = parseFiscalInformation(domesticDutyTaxParties),
+        additionalFiscalReferencesData = parseAdditionalFiscalReferencesData(domesticDutyTaxParties),
+        statisticalValue = parseStatisticalValue(itemXml)
+      )
     }
+
+  private def parseAdditionalFiscalReferencesData(domesticDutyTaxParties: NodeSeq): Option[AdditionalFiscalReferences] = {
+    val references = (domesticDutyTaxParties \ ID).flatMap { reference =>
+      val text = reference.text
+      if (text.length > 1) Some(AdditionalFiscalReference(country = text.take(2), reference = text.drop(2))) else None
```

Comment:
Schema has `an..17` as format for `(domesticDutyTaxParties \ ID)`
I think no harm to have the condition `text.length > 1`, if not let me know.

Reply:
Length check seems unnecessary in this place. Both `take` and `drop` methods return empty String should the index not exist.
And instead of calling simply `text` method on `reference`, I think it would be better to use `StringOption` and work with `Option[String]` type instead.

Reply:
The test was to added to avoid to have a country with 1 character only, since the schema does not specify a minimal length.